tools/update.php

```diff
@@ -0,0 +1,746 @@
+    $db_config['host']
+);
+$paths     = $config->getSetting('paths');
+$loris_root_dir = $paths['base'];
+
+$preupdate_version = getVersionFromLORISRoot($loris_root_dir);
+$info            = json_decode(getLatestReleaseInfo());
+$release_version = $info->{'tag_name'};
+$release_version = substr($release_version, 1); // remove leading 'v'
+
+// Check that the backup directory argument is present and valid
+$backup_dir = $argv[1] ?? '';
+if (!is_dir($backup_dir)) {
+    echo '[-] ERROR: Argument suppled for backup directory is not a valid '
+        . 'directory.' . PHP_EOL;
+    die(usageString());
+}
+// Only apply SQL patches if user explicitly requests it
+$apply_patches = false;
+if (in_array('--apply-patches', $argv)) {
+    $apply_patches = true;
+}
+// Require a confirm flag so this script is not accidentally run
+if (!in_array('--confirm', $argv)) {
+    echo "Your LORIS version is $preupdate_version. The current LORIS version "
+        . "is $release_version. "
+        . PHP_EOL
+        . 'Please run this script with the --confirm flag '
+        . 'if you wish to update your environment and source code to be '
+        . "compatiable with version $release_version."
+        . PHP_EOL;
+    die(usageString());
+}
+// Display some info and wait so the user can read it.
+echo 'This script will prompt for superuser privileges,'
+    . ' as they are needed to e.g. update apt packages.' . PHP_EOL;
+echo 'You may wish to review code changes tagged with "Caveat For '
+    . 'Existing Projects" as they may include changes to your workflow. '
+    . PHP_EOL
+    . "\t" . 'See: https://github.com/aces/Loris/pulls?'
+    . 'q=is%3Apr+label%3A%22Caveat+for+Existing+Projects%22+is%3Amerged'
+    . PHP_EOL . PHP_EOL;
+sleep(3);
+
+/* BEGIN UPDATE PROCESS */
+
+/* Below are all the apt packages required for LORIS to run.
+ *      @see: https://github.com/aces/Loris/wiki/Installing-Loris-in-Depth
+ */
+$loris_requirements = [
+                       "wget",
+                       "zip",
+                       "unzip",
+                       "php-json",
+                       "npm",
+                       "software-properties-common",
+                       "php-ast",
+                       "php$php_version",
+                       "php$php_version-mysql",
+                       "php$php_version-xml",
+                       "php$php_version-json",
+                       "php$php_version-mbstring",
+                       "php$php_version-gd",
+                       "libapache2-mod-php$php_version",
+                      ];
+
+// Update source code.
+echo '[***] Beginning LORIS update process.' . PHP_EOL;
+echo '[*] Release notes:' . PHP_EOL;
+echo $info->{'body'} . PHP_EOL . PHP_EOL;
+echo "[**] Updating LORIS source code "
+    . "($preupdate_version --> $release_version)" . PHP_EOL;
+if (updateSourceCode($loris_root_dir, $backup_dir)) {
+    echo '[+] LORIS source code files successfully updated.' . PHP_EOL;
+}
+
```

Comment:
shouldn't it check the php version, apache version mysql version before starting ?

Reply:
for example 20.0.0 requires apache 2.4 if the update runs without prompting for it, everything is gonna look fine but the website will not be functional

Reply:
Yes this is a good idea.

Reply:
This has been added